ConnectionsPanel.js

```diff
@@ -272,25 +282,52 @@ class ConnectionsPanel extends React.PureComponent {
             }
           }
           if (this.props.connectionsMode === null) {
-            viewList.push(
-              <ResourcesList
-                key={"resourcesList"}
-                relatedHasError={this.props.relatedHasError}
-                sheet={this.props.sheet}
-                themeStr={this.props.themeStr}
-                versionsCount={this.props.versions.length}
-                sheetsCount={this.props.relatedData.sheets ? this.props.relatedData.sheets.length : 0}
-                topicsCount={this.props.relatedData.topics ? Sefaria.links.topicsCount(this.props.relatedData.topics) : 0}
-                setConnectionsMode={this.props.setConnectionsMode}
-                segmentRef={this.props.segmentRef}
-                heSegmentRef={this.props.heSegmentRef}
-                categories={this.props.categories}
-                shareCurrentSegment={this.props.shareCurrentSegment}
-                reportError={this.props.reportError}
-                viewOnSite={this.props.viewOnSite}
-                reloadRelated={this.reloadRelated}
-              />
-            );
+              const collapsedTopLevelLimit = 3;
```

Comment:
(1) the `default` case of the `switch` should be it's own component. There's a lot of logic here.
(2) the new component should rely on a hook `useLinkSummaryList` (not 100% on the name) which stores the state of `showAllRelated` and returns (a) a callback to toggle `showAllRelated`, (b) just the data that needs to be displayed (not the React components). This will separate data from view logic which is currently very mixed and confusing. I realize it was like this prior to your work but your work is building on it.